### PR TITLE
RelocationTools to relocate more Site types

### PR DESCRIPTION
**Pull request by eddieh-xlnx**
Result of this discussion https://github.com/Xilinx/RapidWright/discussions/292#discussioncomment-1661671 as discovered by @bn999.

### CODE REVIEW

**Comment by eddieh-xlnx** [comment]
@clavin-xlnx Even though I think this is sufficient for @bn999, should we add even more types?

**Comment by clavin-xlnx** [comment]
> @clavin-xlnx Even though I think this is sufficient for @bn999, should we add even more types?

I think those changes make sense.  However, from the issue raised in #292, I wonder if we should have a wholesale mode.  That is, an option that relocates all types, which is essentially what @bn999 is trying to do.

**Comment by eddieh-xlnx** [comment]
> I think those changes make sense. However, from the issue raised in #292, I wonder if we should have a wholesale mode. That is, an option that relocates all types, which is essentially what @bn999 is trying to do.

I was hesitant to enable blanket relocation of all types, as some things don't make sense, like I/O and perhaps Laguna sites, and even things like BUFG when we are not making SLR magnitude moves. Failed relocation means that nothing is relocated, whereas right now it just emits a warning.

**Comment by eddieh-xlnx** [comment]
Looks like moving BUFGCEs now causes this error for `com.xilinx.rapidwright.design.TestRelocationTools.testMicroBlazeAndILA`:
```
    ERROR: Failed to move SiteInst 'BUFGCE_X1Y50' from Tile 'XIPHY_L_X32Y120' to Tile 'XIPHY_L_X32Y125'
    ERROR: Failed to move SiteInst 'BUFGCE_X1Y49' from Tile 'XIPHY_L_X32Y120' to Tile 'XIPHY_L_X32Y125'
    ERROR: Failed to move SiteInst 'BUFGCE_X0Y47' from Tile 'XIPHY_L_X0Y60' to Tile 'XIPHY_L_X0Y65'
    ERROR: Failed to move SiteInst 'BUFGCE_X0Y24' from Tile 'XIPHY_L_X0Y60' to Tile 'XIPHY_L_X0Y65'
```

**Comment by eddieh-xlnx** [comment]
TODO: Add a `Set<SiteTypeEnum>` overload to `relocate` functions.

**Comment by eddieh-xlnx** [comment]
Okay, I've added a `siteTypes` argument to the following method:
```
public static boolean relocate(Design design,
                                   String instanceName,
                                   int tileColOffset,
                                   int tileRowOffset,
                                   Set<SiteTypeEnum> siteTypes)
```
allowing you specify your own siteTypes, with a default value of what we have today.

Alternatively for @bn999's original problem Instead of using the `instanceName` overload, you can also use the `PBlock` overload:
```
public static boolean relocate(Design design,
                                   PBlock pblock,
                                   int tileColOffset,
                                   int tileRowOffset)
```
which will attempt to move all site instances inside that PBlock (which can be an entire SLR, e.g. `SLR0:SLR0`) irrespective of type.

File: src/com/xilinx/rapidwright/design/tools/RelocationTools.java [hunk 1]
```diff
@@ -66,12 +79,16 @@
      *                     (empty for top cell)
      * @param tileColOffset Relocate this number of tile columns (X axis)
      * @param tileRowOffset Relocate this number of tile rows (Y axis)
+     * @param siteTypes     Set of SiteTypeEnum-s to relocate
+     *                      (overload exists where this is emitted thus
```
Comment (clavin-xlnx) [inline src/com/xilinx/rapidwright/design/tools/RelocationTools.java:83]:
```suggestion
     *                      (overload exists where this is omitted thus
```

**Review by clavin-xlnx — APPROVED** [review]

**Pull request merged**